### PR TITLE
hid-tmff2: init at v0.81

### PR DESCRIPTION
**Pull request by RaySlash**
## Description

A linux kernel module for Thrustmaster T300RS, T248 and TX(experimental) wheels.
https://github.com/Kimplul/hid-tmff2/commit/ca168637fbfb085ebc9ade0c47fa0653dac5d25b

- Add myself ``rayslash`` as maintainer
- Add hid-tmff2 derivation

> **Note:** Until the updated ``hid-tminit`` is upstreamed, one might want to blacklist the kernel module ``hid-thrustmaster`` to use this module.
 
## Things done

- Built on platform(s):
  - [x] x86_64-linux
  - [ ]  aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is sandbox = true set in nix.conf? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))

- [x] Tested, as applicable:
  - [x] [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - [x] and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - [x] or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - [x] made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages

- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in ./result/bin/)
- [ ] [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
    - [ ] (Package updates) Added a release notes entry if the change is major or breaking
    - [ ] (Module updates) Added a release notes entry if the change is significant
    - [ ] (Module addition) Added a release notes entry if adding a new NixOS module

- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by RaySlash** [comment]
I had to force-push to fix indentation error. My ``nixpkgs-review pr 257386`` output:
```console
[nix-shell:~/nixpkgs]$ nixpkgs-review pr 257386
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/257386/head:refs/nixpkgs-review/1
From https://github.com/NixOS/nixpkgs
 * [new ref]                   refs/pull/257386/head -> refs/nixpkgs-review/1
$ git worktree add /home/smj/.cache/nixpkgs-review/pr-257386/nixpkgs f75e92b46b97df7cc05a1db95ac2607a7207ad1c
Preparing worktree (detached HEAD f75e92b46b97)
Updating files: 100% (37286/37286), done.
HEAD is now at f75e92b46b97 Merge pull request #257506 from Artturin/pinyinfixstrict
$ git merge --no-commit --no-ff 73683e0e6cca42af6437b80c50a93dcea7b01b71
Auto-merging maintainers/maintainer-list.nix
Automatic merge went well; stopped before committing as requested
$ nix build --nix-path nixpkgs=/home/smj/.cache/nixpkgs-review/pr-257386/nixpkgs nixpkgs-overlays=/run/user/1000/tmpu27uotpy --extra-experimental-features nix-command no-url-literals --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed -f /home/smj/.cache/nixpkgs-review/pr-257386/build.nix

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/257386

2 packages marked as broken and skipped:
linuxKernel.packages.hardkernel_4_14.hid-tmff2 linuxPackages_hardkernel_latest.hid-tmff2

23 packages built:
linuxKernel.packages.linux_4_14.hid-tmff2 linuxPackages_4_14_hardened.hid-tmff2 linuxKernel.packages.linux_4_19.hid-tmff2 linuxPackages_4_19_hardened.hid-tmff2 linuxKernel.packages.linux_5_10.hid-tmff2 linuxPackages_5_10_hardened.hid-tmff2 linuxKernel.packages.linux_5_15.hid-tmff2 linuxPackages_5_15_hardened.hid-tmff2 linuxKernel.packages.linux_5_4.hid-tmff2 linuxPackages_5_4_hardened.hid-tmff2 linuxPackages.hid-tmff2 linuxPackages_hardened.hid-tmff2 linuxKernel.packages.linux_6_4.hid-tmff2 linuxPackages_6_4_hardened.hid-tmff2 linuxPackages_latest.hid-tmff2 linuxPackages_6_5_hardened.hid-tmff2 linuxPackages_latest-libre.hid-tmff2 linuxPackages-libre.hid-tmff2 linuxPackages_lqx.hid-tmff2 linuxPackages_testing_bcachefs.hid-tmff2 linuxPackages_xanmod.hid-tmff2 linuxPackages_xanmod_latest.hid-tmff2 linuxPackages_zen.hid-tmff2

error: cannot find flake 'flake:None' in the flake registries
error: cannot find flake 'flake:None' in the flake registries
$ /nix/store/ggza64x1frk77rcyzlpc3nwwp5v726vf-nix-2.17.0/bin/nix-shell /home/smj/.cache/nixpkgs-review/pr-257386/shell.nix --nix-path nixpkgs=/home/smj/.cache/nixpkgs-review/pr-257386/nixpkgs nixpkgs-overlays=/run/user/1000/tmpu27uotpy
```

File: pkgs/os-specific/linux/hid-tmff2/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+stdenv.mkDerivation {
+  pname = "hid-tmff2";
+  version = "unstable-2023-09-23";
+
+  src = fetchFromGitHub {
+    owner = "Kimplul";
+    repo = "hid-tmff2";
+    rev = "b2872a893afd0f9cd0bfd4644348f3b8645edbed";
+    hash = "sha256-7AlBxFUqw0nFl9f03WWDhUQsd1RCdtvcUeVWfcO9J8Q=";
+    fetchSubmodules = true;
```
Comment (Artturin) [inline pkgs/os-specific/linux/hid-tmff2/default.nix:12]:
```suggestion
    # for hid-tminit
    fetchSubmodules = true;
```

**Comment by Artturin** [comment]
When doing fixup commits please fixup/squash

`#3` here https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions

**Comment by Artturin** [comment]
Please do not have merge commits in the pr, if there's conflicts then rebase.

**Comment by RaySlash** [comment]
sorry, ill rebase and try this time. I am pretty new to contributing into huge repos like nixpkgs.

**Comment by RaySlash** [comment]
Hey,

- The comment above `fetchSubmodules`[[hid-tmff2/default.nix#L12]](https://github.com/RaySlash/nixpkgs/blob/0fdce2de590b9bbf6bf079e5c76998f296dea7a2/pkgs/os-specific/linux/hid-tmff2/default.nix#L12) have been updated to reflect on `hid-tminit`
- The driver version is now updated as the maintainer agreed on a versioning system as well. The version is updated to currently `v0.81` as per it reflects [here](https://github.com/Kimplul/hid-tmff2/blob/ca168637fbfb085ebc9ade0c47fa0653dac5d25b/dkms/dkms-install.sh#L12).

Sorry for the force-pushes. I had messed up previously with merging from master which is now resolved. Let me know if you would like to see this as a new PR in case this doesn't meet the requirements.

File: pkgs/os-specific/linux/hid-tmff2/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+stdenv.mkDerivation {
+  pname = "hid-tmff2";
+  version = "v0.81";
+
+  src = fetchFromGitHub {
+    owner = "Kimplul";
+    repo = "hid-tmff2";
+    rev = "ca168637fbfb085ebc9ade0c47fa0653dac5d25b";
+    hash = "sha256-7AlBxFUqw0nFl9f03WWDhUQsd1RCdtvcUeVWfcO9J8Q=";
```
Comment (Artturin) [inline pkgs/os-specific/linux/hid-tmff2/default.nix:11]:
```
error: hash mismatch in fixed-output derivation '/nix/store/l58rmv4nv7wd27fx1hnqaq42zkl3km6l-source.drv':
         specified: sha256-7AlBxFUqw0nFl9f03WWDhUQsd1RCdtvcUeVWfcO9J8Q=
            got:    sha256-Nm5m5xjwJGy+ia4nTkvPZynIxUj6MVGGbSNmIcIpziM=
```

**Comment by Artturin** [comment]
Fixup the fixups then it should be ok

**Comment by Artturin** [comment]
Congratulations on your first PR

**Pull request merged**